x/emissions/keeper/msgserver/msg_server_params.go

```diff
@@ -128,6 +119,27 @@ func (ms msgServer) UpdateParams(ctx context.Context, msg *types.MsgUpdateParams
 	if len(newParams.BlocksPerMonth) == 1 {
 		existingParams.BlocksPerMonth = newParams.BlocksPerMonth[0]
 	}
+	if len(newParams.PRewardInference) == 1 {
+		existingParams.PRewardInference = newParams.PRewardInference[0]
+	}
+	if len(newParams.PRewardForecast) == 1 {
+		existingParams.PRewardForecast = newParams.PRewardForecast[0]
+	}
+	if len(newParams.PRewardReputer) == 1 {
+		existingParams.PRewardReputer = newParams.PRewardReputer[0]
+	}
+	if len(newParams.CRewardInference) == 1 {
+		existingParams.CRewardInference = newParams.CRewardInference[0]
+	}
+	if len(newParams.CRewardForecast) == 1 {
+		existingParams.CRewardForecast = newParams.CRewardForecast[0]
+	}
+	if len(newParams.FTolerance) == 1 {
+		existingParams.FTolerance = newParams.FTolerance[0]
+	}
+	if len(newParams.CNorm) == 1 {
+		existingParams.CNorm = newParams.CNorm[0]
+	}
 	err = existingParams.Validate()
```

Comment:
Need to update this function to do proper validation on the new params

Reply:
Added inside the validate function